test/distributed/checkpoint/test_torch_save_to_dcp.py

```diff
@@ -0,0 +1,248 @@
+import copy
+import logging
+import os
+import tempfile
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+import torch.distributed as dist
+import torch.distributed.checkpoint as DCP
+import torch.nn as nn
+from torch.distributed._shard._utils import narrow_tensor_by_index
+from torch.distributed._tensor import DTensor, Placement, Replicate, Shard
+from torch.distributed.checkpoint.filesystem import FileSystemReader
+from torch.distributed.checkpoint.metadata import (
+    ChunkStorageMetadata,
+    Metadata,
+    TensorProperties,
+    TensorStorageMetadata,
+)
+from torch.distributed.checkpoint.planner import LoadItemType, LoadPlan, LoadPlanner
+from torch.distributed.checkpoint.serialization import Deserializer, Serializer
+from torch.distributed.checkpoint.storage import StorageReader, StorageWriter
+from torch.distributed.device_mesh import init_device_mesh
+from torch.futures import Future
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    skip_if_lt_x_gpu,
+    with_comms,
+)
+from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+
+
+class TestDummyModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        torch.manual_seed(0)
+        self.net1 = nn.Linear(8, 16)
+        self.net2 = nn.Linear(16, 32)
+        self.net3 = nn.Linear(32, 64)
+        self.net4 = nn.Linear(64, 8)
+
+    def forward(self, x):
+        x = F.relu(self.net1(x))
+        x = F.relu(self.net2(x))
+        x = F.relu(self.net3(x))
+        x = F.relu(self.net4(x))
+        return x
+
+
+def _construct_sharded_state_dict(fp_name: str, world_size: int) -> Dict[str, Any]:
```

Comment:
seems odd to me that this util wouldn't already exist somewhere in the real dcp code or as a torch util.  should it be one (and any opportunity to replace existing code with it?)

Reply:
Yup, I agree this should be an util in DCP.